src/control-plane/backend/insert-admin-user.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { join } from 'path';
+import { Duration, CustomResource, Stack } from 'aws-cdk-lib';
+import { Table } from 'aws-cdk-lib/aws-dynamodb';
+import { Effect, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Runtime, Function } from 'aws-cdk-lib/aws-lambda';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { RetentionDays } from 'aws-cdk-lib/aws-logs';
+import { Provider } from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+import { addCfnNagToStack, ruleForLambdaVPCAndReservedConcurrentExecutions } from '../../common/cfn-nag';
+import { createLambdaRole } from '../../common/lambda';
+import { attachListTagsPolicyForFunction } from '../../common/lambda/tags';
+import { POWERTOOLS_ENVS } from '../../common/powertools';
+
+export interface AddAdminUserProps {
+  readonly email: string;
+  readonly userTable: Table;
+}
+
+export class AddAdminUser extends Construct {
+
+  readonly crForAddAdminUser: CustomResource;
+
+  constructor(scope: Construct, id: string, props: AddAdminUserProps) {
+    super(scope, id);
+
+    this.crForAddAdminUser = this.createAddAdminUserCustomResource(props);
+  }
+
+  private createAddAdminUserCustomResource(props: AddAdminUserProps): CustomResource {
```

Comment:
Using too much sketching code for making a simple API call, try to use CDK's `AwsCustomResource` to simplify it

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources-readme.html#custom-resources-for-aws-apis